libs/payments/src/lib/payment-provider/bexioPaymentProvider.ts

```diff
@@ -0,0 +1,237 @@
+    this.invoiceTemplateRenewalMembership = props.invoiceTemplateRenewalMembership
+    this.unitId = props.unitId
+    this.taxId = props.taxId
+    this.accountId = props.accountId
+    this.invoiceMailSubject = props.invoiceMailSubject
+    this.invoiceMailBody = props.invoiceMailBody
+    this.markInvoiceAsOpen = props.markInvoiceAsOpen
+    this.prisma = props.prisma
+  }
+
+  async webhookForPaymentIntent(props: WebhookForPaymentIntentProps): Promise<IntentState[]> {
+    throw Error('Webhook not implemented!')
+  }
+
+  /**
+   * Create remote invoice only in case of first
+   * @param props
+   */
+
+  async createIntent(props: CreatePaymentIntentProps): Promise<Intent> {
+    const subscription = await this.prisma.subscription.findUnique({
+      where: {id: props.invoice.subscriptionID}
+    })
+    if (!subscription.paidUntil) {
+      return await this.bexioCreate(props.invoice.id, this.invoiceTemplateNewMembership)
+    }
+  }
+
+  /**
+   * Create remote invoice in case of renewal
+   * @param props
+   */
+
+  async createRemoteInvoice(props: CreateRemoteInvoiceProps) {
+    await this.bexioCreate(props.invoice.id, this.invoiceTemplateRenewalMembership)
+  }
+
+  async checkIntentStatus({intentID}: CheckIntentProps): Promise<IntentState> {
+    return {
+      state: PaymentState.requiresUserAction,
+      paidAt: null,
+      paymentID: 'None',
+      paymentData: ''
+    }
+  }
+
+  async bexioCreate(invoiceId: string, template: string) {
+    try {
+      const invoice = await this.prisma.invoice.findUnique({
+        where: {
+          id: invoiceId
+        },
+        include: {
+          subscription: {
+            include: {
+              memberPlan: true,
+              user: {
+                include: {
+                  address: true
+                }
+              }
+            }
+          }
+        }
+      })
+      const bexio = new Bexio(this.apiKey)
+      const contact = await this.searchForContact(bexio, invoice.subscription.user)
+      const updatedContact = await this.createOrUpdateContact(
+        bexio,
+        contact,
+        invoice.subscription.user
+      )
+      await this.createInvoice(bexio, updatedContact, invoice, template)
+      return {
+        intentID: '',
+        intentSecret: '',
+        intentData: '',
+        paidAt: null,
+        state: PaymentState.created
+      }
+    } catch (e) {
+      console.error(e)
+      throw e
+    }
+  }
+
+  async searchForContact(bexio: Bexio, user: User & {address: UserAddress}) {
+    const contacts = await bexio.contacts.search([
+      {
+        field: ContactSearchParameters.mail,
+        value: `${user.email}`,
+        criteria: '='
+      }
+    ])
+    return contacts[0]
+  }
+
+  async createOrUpdateContact(
+    bexio: Bexio,
+    contact: ContactsStatic.ContactSmall,
+    user: User & {address: UserAddress}
+  ) {
+    const uppserContact: ContactsStatic.ContactOverwrite = {
```

Comment:
do you mean upsert?

Reply:
Yes wasn't finished yet but had a call and forgot to set it to do draft. But thanks, changed this.